### ISSUE TITLE
Bug: Unexpected Token error for valid class field declarations

### ISSUE THREAD

**Issue by TotallyInformation** [eslint/eslint#19430]
### Environment

Node version: v20.18.1
npm version: N/A (running in VS Code)
Local ESLint version:  9.20.1
Global ESLint version: None
Operating System: Windows 11

Note that the error happens with other eslint versions. Tested with at least v6 as well.

### What parser are you using?

Default (Espree)

### What did you do?

Attempted to reproduce the error in the demo but it does not appear. Note that the error occurs both in VS Code and when running the local ESLINT via the command line on the exact code shown below.

<details>
<summary>Configuration</summary>

```
/**
 * https://www.npmjs.com/search?q=eslint-config
 * https://www.npmjs.com/search?q=keywords:eslint
 * 
 * npm init @eslint/config@latest -- --config eslint-config-standard
 * https://eslint.org/docs/latest/rules
 */

import globals from 'globals' // https://www.npmjs.com/package/globals
// @ts-ignore
import importPlugin from 'eslint-plugin-import' // https://www.npmjs.com/package/eslint-plugin-import
import pluginPromise from 'eslint-plugin-promise' // https://www.npmjs.com/package/eslint-plugin-promise
import jsdoc from 'eslint-plugin-jsdoc'// https://github.com/gajus/eslint-plugin-jsdoc
// import nodePlugin from 'eslint-plugin-n' // https://www.npmjs.com/package/eslint-plugin-n, node.js only
import pluginJs from '@eslint/js'


/** @type {import('eslint').Linter.Config[]} */
export default [
    {
        // files: ['**/*.{js,mjs,cjs}'],
        languageOptions: {
            sourceType: 'module',
            ecmaVersion: 'latest',
            globals: globals.browser,
        },

        rules: {
            'prefer-const': 'error',
        },
    },
    pluginJs.configs.recommended,
    jsdoc.configs['flat/recommended'],
    pluginPromise.configs['flat/recommended'],
    importPlugin.flatConfigs.recommended,
    // nodePlugin.configs["flat/recommended-script"], // Node.js only
]

```
</details>

```js
class VisibleConsole extends HTMLElement {
    /** Component version */
    static componentVersion = '2025-02-12'
}
```

Error shown when run on command line:

```
> npm run lint

> @totallyinformation/web-components@1.0.0 lint
> eslint ./test.js


D:\src\web-components\test.js
  3:29  error  Parsing error: Unexpected token =

✖ 1 problem (1 error, 0 warnings)
```

### What did you expect to happen?

No error should be shown since the syntax is indeed valid modern (since at least 2015) JavaScript for class definitions.

The class works perfectly with this code.

### What actually happened?

VS Code gives an error which cannot be corrected:

![Image](https://github.com/user-attachments/assets/3bc8a621-26b8-4997-96c9-6426bd4e8756)

From the command line:

![Image](https://github.com/user-attachments/assets/2217a97f-f133-4cb7-aac8-b96e56f0f6da)

It also appears as though no other linting is happening for the file. As seen in this example where at least 2 other errors should be visible:

![Image](https://github.com/user-attachments/assets/e1dbe738-4d61-4b6b-bf46-24eda07cb1f3)

### Link to Minimal Reproducible Example

https://github.com/TotallyInformation/web-components

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

In the given example link. `./test.js` contains the example code. `./esbuild.config.mjs` contains the config.

**Comment by nzakas** [comment]
One of your other configs is resetting `ecmaVersion` to 2018

```shell
@nzakas ➜ /workspaces/web-components (main) $ npx eslint --print-config test.js | grep ecmaVersion
    "ecmaVersion": 2018,
```

See https://eslint.org/docs/latest/use/configure/debug
